### ISSUE TITLE
Conflict with horizon

### ISSUE THREAD

**Issue by gdluck** [jenssegers/laravel-mongodb#3060]
- Laravel-mongodb Version: #.#.#
- PHP Version: 8.3
- Database Driver & Version: latest horizon, latest laravel-mongodb, latest laravel

### Description:

Your service provider conflicts with horizon: 
https://github.com/laravel/horizon/issues/1476

Works only if I comment the BatchRepository::class in MongoDBBusServiceProvider.php
![photo_2024-07-22_14-55-29](https://github.com/user-attachments/assets/c77601a4-bf40-4ba7-8eab-cd102401b6d9)

**Comment by alcaeus** [comment]
Hi @gdluck, neither this ticket nor the linked one contain any steps for us to reproduce, and from the error message I can't see how this package resulted in this. Please add steps to reproduce and include the full exception details, including the stack trace, in your report. Thank you.

**Comment by boat-ji** [comment]
If you are not using MongoDB Batches, downgrading to version 4.3.1 should resolve the issue. I faced a similar problem, and this downgrade worked for me.

**Comment by GromNaN** [comment]
@BoatFazWaz We'd like you to be able to update the package, but to do so we need precise information about the problem. Ideally a small Laravel project that reproduces the issue, so we can work on it.

**Comment by gdluck** [comment]
Steps to reproduce: 

composer create-project laravel/laravel testapp

modify .env from sqlite to mysql with correct infos and create an empty database: mysql -Bse 'create database laravel'

DB_CONNECTION=mysql
DB_HOST=127.0.0.1
DB_PORT=3306
DB_DATABASE=laravel
DB_USERNAME=root
DB_PASSWORD=

php artisan migrate:install
php artisan migrate:refresh
composer require laravel/horizon
composer require mongodb/laravel-mongodb:^4.7

php artisan horizon:install

in ./app/Providers/HorizonServiceProvider.php add in gate "return true;"

php artisan horizon &
php artisan serve

then access:
http://127.0.0.1:8000/horizon/batches

and you will get the 500 error 

After changing any file from vendor, run - php artisan optimize:clear

<img width="1218" alt="Screenshot 2024-07-23 at 12 07 49" src="https://github.com/user-attachments/assets/ec5f24b7-b5fc-439b-8083-8a9665d0c34d">

**Comment by degecko** [comment]
The way you're extending the BatchRepository provider is causing some sort of double-resolving issue.

**Comment by bisht2050** [comment]
Tracked with [PHPORM-222](https://jira.mongodb.org/browse/PHPORM-222).

**Comment by GromNaN** [comment]
Could you confirm this fix is correct: https://github.com/mongodb/laravel-mongodb/pull/3071

**Comment by gdluck** [comment]
I've edited the comment with detailed steps to reproduce the error. I can confirm that the issue is now resolved.

**Comment by polatdev** [comment]
> I've edited the comment with detailed steps to reproduce the error. I can confirm that the issue is now resolved.

how to resolve?

**Comment by GromNaN** [comment]
The immediate solution it to register the service in your application. If you use MongoDB for queue:

```php
$this->app->singleton(BatchRepository::class, function (Container $app) {
    return $app->make(MongoBatchRepository::class);
});
```

Or if you use the database driver, copy [from `BusServiceProvider`](https://github.com/laravel/framework/blob/b18f33340ca5a691132bccb3ad5de59c7fbdfdd9/src/Illuminate/Bus/BusServiceProvider.php#L46-L52).

**Comment by GromNaN** [comment]
Fixed in version [4.7.1](https://github.com/mongodb/laravel-mongodb/releases/tag/4.7.1) by https://github.com/mongodb/laravel-mongodb/pull/3071

**Comment by gdluck** [comment]
Thank you!
